standardization/src/main/scala/za/co/absa/enceladus/standardization/StandardizationJob.scala

```diff
@@ -128,45 +130,94 @@ object StandardizationJob {
     spark
   }
 
-  private def readerFormatSpecific(dfReader: DataFrameReader, cmd: CmdConfig, dataset: Dataset): DataFrameReader = {
+  /**
+    * Applies all format-specific options provided by command line parameters to a dataframe reader.
+    *
+    * @param dfReader An input dataframe reader
+    * @param cmd      Command line parameters containing format-specific options
+    * @param dataset  A dataset definition
+    * @return The updated dataframe reader
+    */
+  def getFormatSpecificReader(dfReader: DataFrameReader, cmd: CmdConfig, dataset: Dataset): DataFrameReader = {
     // applying format specific options
-    val dfReaderWithOptions = {
-      val dfr1 = if (cmd.rowTag.isDefined) dfReader.option("rowTag", cmd.rowTag.get) else dfReader
-      val dfr2 = if (cmd.csvDelimiter.isDefined) dfr1.option("delimiter", cmd.csvDelimiter.get) else dfr1
-      val dfr3 = if (cmd.csvHeader.isDefined) dfr2.option("header", cmd.csvHeader.get) else dfr2
-      val dfr4 = if (cmd.rawFormat.equalsIgnoreCase("cobol")) applyCobolOptions(dfr3, cmd, dataset) else dfr3
-      dfr4
-    }
-    if (cmd.rawFormat.equalsIgnoreCase("fixed-width")) {
-      val dfWithFixedWithOptions = if (cmd.fixedWidthTrimValues.get) {
-        dfReaderWithOptions.option("trimValues", "true")
-      } else {
-        dfReaderWithOptions
+    val options = getCobolOptions(cmd, dataset) ++
+      getGenericOptions(cmd) ++
+      getXmlOptions(cmd) ++
+      getCsvOptions(cmd) ++
+      getFixedWidthOptions(cmd)
+
+    // Applying all the options
+    options.foldLeft(dfReader) { (df, optionPair) =>
+      optionPair match {
+        case (key, Some(value)) =>
```

Comment:
What do you think about this code:
```
        case (key, Some(s: String)) => df.option(key, s)
        case (key, Some(b: Boolean)) => df.option(key, b)
        case (key, Some(l: Boolean)) => df.option(key, l)
        case (key, Some(d: Boolean)) => df.option(key, d)
        case _ => df
```
Also have the additional benefit, that it will not crash if any of the parameter values is not `String`, `Boolean`, `Long` nor `Double`. Which IMO the original code would.
I would still consider adding some type conversion Int->Long etc.

Reply:
The old code won't crash, true. But it would fail fail on a compilation error if a value of an unsupported type is passed into one of .option() overloads. We are trading compile time error versus runtime error. Still I want this to crash instead of silently ignoring an option of an unrecognized type.
Will add Int conversion though.

Reply:
👍 Reimplemented it using the strict typing approach you have suggested.

Reply:
Thank you 🙇